src/log.h

```diff
@@ -27,6 +27,8 @@ struct raft_entry_ref
     raft_term term;              /* Term of the entry being ref-counted. */
     raft_index index;            /* Index of the entry being ref-counted. */
     unsigned short count;        /* Number of references. */
+    struct raft_buffer buf;      /* Private buffer for the tracked entry */
```

Comment:
The comment is descriptive, but it would probably be more useful to add the rationale why `buf` and `batch` are added.